src/dosext/dpmi/dpmi.c

```diff
@@ -1554,6 +1554,18 @@ void dpmi_set_interrupt_vector(unsigned char num, DPMI_INTDESC desc)
 {
     DPMI_CLIENT.Interrupt_Table[num].selector = desc.selector;
     DPMI_CLIENT.Interrupt_Table[num].offset = desc.offset32;
+    if (config.cpu_vm_dpmi == CPUVM_KVM) {
+        /* KVM: we can directly use the IDT but don't do it when debugging */
+#ifdef USE_MHPDBG
+        if (mhpdbg.active && dpmi_mhp_intxxtab[num])
+            kvm_set_idt_default(num);
+        else
+#endif
+        if (desc.selector == dpmi_sel())
+            kvm_set_idt_default(num);
+        else
+            kvm_set_idt(num, desc.selector, desc.offset32, DPMI_CLIENT.is_32);
```

Comment:
Mm, are you sure of that?
Maybe we should check is_32 of
the handler code selector, like
`Segments[desc.selector>>3].is_32`?

Reply:
It's consistent with `dpmi_iret` this way where it was causing issues without my last change. I would need to check dpmi to see what is correct, but it needs to be consistent for sure.

Reply:
It's consistent with `dpmi_iret` this way where it was causing issues without my last change. I would need to check dpmi to see what is correct, but it needs to be consistent for sure.

Reply:
I think DPMI_CLIENT.is_32 is correct, from http://www.delorie.com/djgpp/doc/dpmi/api/310205.html: "32-bit clients must supply a 32-bit offset in the EDX register. If the client's handler chains to the next exception handler it must do so using a 32-bit interrupt stack frame."

Checking the desc.selector for 32-bit would contradict this if it's a 16-bit routine with a 32-bit stack.